sandbox/host/src/sandbox.rs

```diff
@@ -243,11 +251,55 @@ impl SandboxInstance {
         value: Value,
     ) -> std::result::Result<Option<()>, error::Error> {
         match &self.backend_instance {
-            BackendInstance::Wasmi(wasmi_instance) => wasmi_set_global(wasmi_instance, name, value),
+            BackendInstanceBundle::Wasmi(wasmi_instance) => {
+                wasmi_set_global(wasmi_instance, name, value)
+            }
+
+            BackendInstanceBundle::Wasmer { instance, store } => {
+                wasmer_set_global(instance, &mut store.borrow_mut(), name, value)
+            }
+        }
+    }
+
+    /// Get the value from a global with the given `name`. Only for usage in signal handler.
+    ///
+    /// Returns `Some(_)` if the global could be found.
+    ///
+    /// # Safety
+    ///
+    /// Expected to be called only from signal handler.
+    pub unsafe fn signal_handler_get_global_val(&self, name: &str) -> Option<Value> {
+        match &self.backend_instance {
+            BackendInstanceBundle::Wasmi(wasmi_instance) => wasmi_get_global(wasmi_instance, name),
 
-            BackendInstance::Wasmer(wasmer_instance) => {
-                wasmer_set_global(wasmer_instance, name, value)
+            BackendInstanceBundle::Wasmer { instance, store } => unsafe {
+                // We cannot use `store.borrow_mut()` in single handler context because it's already borrowed during `invoke` call.
```

Comment:
```suggestion
                // We cannot use `store.borrow_mut()` in signal handler context because it's already borrowed during `invoke` call.
```

Reply:
The same in the next func too